src/FSharp.Core/array.fs

```diff
@@ -1578,19 +1578,38 @@ module Array =
         checkNonNull "array" array
         Microsoft.FSharp.Primitives.Basics.Array.permute indexMap array
 
-    [<CompiledName("Sum")>]
-    let inline sum (array: ^T array) : ^T =
+    let inline private classicSum (array: ^T array) : ^T =
         checkNonNull "array" array
         let mutable acc = LanguagePrimitives.GenericZero< ^T>
 
         for i = 0 to array.Length - 1 do
             acc <- Checked.(+) acc array.[i]
 
         acc
```

Comment:
This will now become non-inlineable, right?

Reply:
Nvm. Disregard the question.

Reply:
Is there a way to get it inlineable and not copy&paste every type-branch to have a separate implementation?